lib/extras/enc/apng.cc

```diff
@@ -440,6 +459,26 @@ Status APNGEncoder::EncodePackedPixelFileToAPNG(
   return true;
 }
 
+Status APNGEncoder::EncodeExtraChannelToPNG(const PackedImage& image,
+                                            size_t bits_per_sample,
+                                            std::vector<uint8_t>* bytes) const {
+  // Create a new PackedImage and move data from the original image
+  PackedImage moved_image(std::move(const_cast<PackedImage&>(image)));
```

Comment:
Since PNG encoding is used in a few other places than cjxl, it might have surprising effects (currently or in the future) if the const & ppf loses its extra channels, so it might be better to copy here. Or alternatively one could add extra arguments to EncodePackedPixelFileToAPNG to select which extra channel/frame to include in the output PNG.

Reply:
I added extra arguments to `EncodePackedPixelFileToAPNG`.